plugin-colorpicker/colorpicker.cpp

```diff
@@ -44,37 +71,76 @@ ColorPicker::~ColorPicker() = default;
 
 void ColorPicker::realign()
 {
-    mWidget.button()->setFixedHeight(panel()->iconSize());
-    mWidget.button()->setFixedWidth(panel()->iconSize());
-    mWidget.lineEdit()->setFixedHeight(panel()->iconSize());
+    mWidget.update(panel()->lineCount() <= 1 ? panel()->isHorizontal() : !panel()->isHorizontal());
 }
 
 
-ColorPickerWidget::ColorPickerWidget(QWidget *parent):
-    QFrame(parent)
+ColorPickerWidget::ColorPickerWidget(QWidget *parent) : QWidget(parent)
 {
-    QFontMetrics fm (mLineEdit.font());
-    mLineEdit.setFixedWidth ( 10*fm.horizontalAdvance (QStringLiteral("a")) );
+    setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
 
-    QHBoxLayout *layout = new QHBoxLayout(this);
-    layout->setContentsMargins (0, 0, 0, 0);
-    layout->setSpacing (1);
-    setLayout(layout);
-    layout->addWidget (&mButton);
-    layout->addWidget (&mLineEdit);
+    mSeparator = new QFrame();
+    mSeparator->setFrameShape(QFrame::VLine);
+    mSeparator->setFrameShadow(QFrame::Sunken);
+    mSeparator->setLineWidth(1);
+    mSeparator->setFixedHeight(16);
 
+    mPickerButton = new QToolButton();
+    mPickerButton->setObjectName(QStringLiteral("ColorPickerPickerButton"));
+    mPickerButton->setAccessibleName(mPickerButton->objectName());
+    mPickerButton->setAutoRaise(true);
+    mPickerButton->setIcon(QIcon::fromTheme(QLatin1String("color-select-symbolic"), QIcon::fromTheme(QLatin1String("color-picker"))));
```

Comment:
Although an icon set like Breeze has "xyz-symbolic" icons, they're mainly for Gnome. IMHO, it's better to reverse the fallback and main icons here.

Reply:
Done